src/rez/utils/execution.py

```diff
@@ -59,15 +58,34 @@ def __init__(self, args, **kwargs):
             if file_no not in (0, 1, 2):
                 kwargs["stdin"] = subprocess.PIPE
 
+        self._set_text_encoding_kwargs(kwargs)
+        super(Popen, self).__init__(args, **kwargs)
+
+    def _set_text_encoding_kwargs(self, kwargs):
+        """ Adds support for py3 :py:obj:`subprocess.Popen`
+        keywords `text`, `encoding` for more consistent handling of stdout/stderr.
+
+        Args:
+            kwargs (dict): keyword arguments for :py:obj:`subprocess.Popen`
+        """
         # Add support for the new py3 "text" arg, which is equivalent to
         # "universal_newlines".
         # https://docs.python.org/3/library/subprocess.html#frequently-used-arguments
         #
-        if "text" in kwargs:
-            kwargs["universal_newlines"] = True
-            del kwargs["text"]
+        text = kwargs.pop('text', None)
+        universal_newlines = kwargs.pop('universal_newlines', None)
+        if not any((text, universal_newlines)):
+            return
+        kwargs["universal_newlines"] = True
 
-        super(Popen, self).__init__(args, **kwargs)
+        # fixes py3/cmd.exe UnicodeDecodeError() with some characters.
+        #    UnicodeDecodeError: 'charmap' codec can't decode byte
+        #    0x8d in position 1023172: character maps to <undefined>
+
+        # NOTE: currently no solution for `python3+<3.6`
+        #
+        if sys.version_info[:2] >= (3, 6):
```

Comment:
I would do this only if `encoding` is not already in kwargs

Reply:
oh thank you! fixing now!